cmd/dockerd/hack/malformed_host_override.go

```diff
@@ -60,27 +60,12 @@ func (l *MalformedHostHeaderOverrideConn) Read(b []byte) (n int, err error) {
 				return c, nil
 			}
 
-			if b[i+1] != 'H' {
-				continue
-			}
-			if b[i+2] != 'o' {
-				continue
-			}
-			if b[i+3] != 's' {
-				continue
-			}
-			if b[i+4] != 't' {
-				continue
-			}
-			if b[i+5] != ':' {
-				continue
-			}
-			if b[i+6] != ' ' {
-				continue
-			}
-			if b[i+7] != '/' {
+			if b[i+1] != 'H' || b[i+2] != 'o' || b[i+3] != 's' ||
```

Comment:
Although it's more _compact_, I don't think this makes it more _readable_ than the existing code. Perhaps something like this would work;

```suggestion
			if string(b[i:i+7]) != "\nHost: " {
```

Reply:
done.